### PR TITLE
remove some redundant code

### PR DESCRIPTION
**Pull request by moving-bits**
- remove unnecessary type identifiers which can be inferred by return / parameter type
- merge duplicate switch branches
- use `Arrays.fill` instead of loop resetting each value manually
- use `Math.min` instead of manual comparison
- remove duplicate string expression
- remove unnecessary null check

### CODE REVIEW

File: main/src/cgeo/geocaching/EditWaypointActivity.java [hunk 1]
```diff
@@ -611,7 +611,6 @@ public void handleMessage(final Message msg) {
                     ActivityMixin.showApplicationToast(activity.getString(R.string.waypoint_coordinates_has_been_modified_on_website, coords));
                     break;
                 case SUCCESS:
-                    break;
```
Comment (bekuno) [inline main/src/cgeo/geocaching/EditWaypointActivity.java:614]: Please do not remove.
The code is clearer with the break statement.

File: main/src/cgeo/geocaching/apps/AbstractLocusApp.java [hunk 2]
```diff
@@ -352,13 +350,12 @@ private static String toLocusWaypoint(final WaypointType wt) {
             case FINAL:
                 return GeocachingWaypoint.CACHE_WAYPOINT_TYPE_FINAL;
             case OWN:
+            case STAGE:
                 return GeocachingWaypoint.CACHE_WAYPOINT_TYPE_PHYSICAL_STAGE;
             case PARKING:
                 return GeocachingWaypoint.CACHE_WAYPOINT_TYPE_PARKING;
             case PUZZLE:
```
Comment (bekuno) [inline main/src/cgeo/geocaching/apps/AbstractLocusApp.java:357]: I would let it as is.
The code is for me clearer then, the case values are sorted alphabeticly.

File: main/src/cgeo/geocaching/MainActivity.java [hunk 3]
```diff
@@ -469,10 +469,7 @@ public boolean onPrepareOptionsMenu(final Menu menu) {
     public boolean onOptionsItemSelected(final MenuItem item) {
         final int id = item.getItemId();
         switch (id) {
-            case android.R.id.home:
-                // this activity must handle the home navigation different than all others
-                showAbout(null);
-                return true;
+            case android.R.id.home: // this activity must handle the home navigation different than all others
```
Comment (bekuno) [inline main/src/cgeo/geocaching/MainActivity.java:472]: I would let it as is.

File: main/src/cgeo/geocaching/connector/su/SuParser.java [hunk 4]
```diff
@@ -345,13 +345,11 @@ private static LogType parseLogType(final String status) {
...
                 return LogType.FOUND_IT;
             case "2":
+            case "4":
                 return LogType.DIDNT_FIND_IT;
             case "3":
                 return LogType.NOTE;
-            case "4":
-                return LogType.DIDNT_FIND_IT;
             case "5":
-                return LogType.OWNER_MAINTENANCE;
```
Comment (bekuno) [inline main/src/cgeo/geocaching/connector/su/SuParser.java:354]: Please do not change, let it sorted.

File: main/src/cgeo/geocaching/connector/su/SuParser.java [hunk 5]
```diff
@@ -361,8 +359,6 @@ private static LogType parseLogType(final String status) {
 
     private static CacheSize parseSize(final String size) {
         switch (size) {
-            case "1":
-                return CacheSize.UNKNOWN;
             case "2":
                 return CacheSize.MICRO;
```
Comment (bekuno) [inline main/src/cgeo/geocaching/connector/su/SuParser.java:363]: Please do not change, let it sorted.

File: main/src/cgeo/geocaching/connector/su/SuParser.java [hunk 6]
```diff
@@ -421,22 +417,16 @@ private static CacheType parseType(final int type) {
...
-                // Virtual Multi-step
-                return CacheType.VIRTUAL;
-            case 8:
-                // Contest
-                return CacheType.EVENT;
             case 9:
-                return CacheType.MYSTERY;
-            case 10:
-                // Mystery Virtual
+            case 10: // Mystery Virtual
```
Comment (bekuno) [inline main/src/cgeo/geocaching/connector/su/SuParser.java:429]: Please do not change, let it sorted.

File: main/src/cgeo/geocaching/utils/AngleUtils.java [hunk 7]
```diff
@@ -46,14 +46,13 @@ public static float normalize(final float angle) {
...
         switch (WindowManagerHolder.WINDOW_MANAGER.getDefaultDisplay().getRotation()) {
-            case Surface.ROTATION_0:
-                return 0;
             case Surface.ROTATION_90:
                 return 90;
             case Surface.ROTATION_180:
                 return 180;
             case Surface.ROTATION_270:
                 return 270;
+            case Surface.ROTATION_0:
```
Comment (bekuno) [inline main/src/cgeo/geocaching/utils/AngleUtils.java:55]: Please let it sorted.

**Comment by moving-bits** [comment]
Rereading this PR I think it mixes too many different optimizations into one PR. I will close this one, split it up into theme-specific, and will leave out the case merging as requested.

**Pull request closed without merging**